src/site/cpp-arrays.rst

```diff
@@ -23,6 +23,129 @@ C++ Lessons #3 --- Arrays
 Static Arrays
 =============
 
+* *Static* arrays mean that the **size** is known at compile time and is not going to change
+* In the below example, we are setting aside a chunk of memory that is ready to store 5 integers
+* Unlike Java however, there are no default values set in the array
+
+.. code-block:: cpp
+    :linenos:
+
+    int foo[5];
+
+
+.. image:: img/cpp_static_array.png
+   :width: 400 px
+   :align: center
+   :target: http://www.cplusplus.com/doc/tutorial/arrays/
+
+* We can also assign values to the array on creation
+
+
+.. code-block:: cpp
+    :linenos:
+
+    int foo[] = {16, 2, 77, 40, 12071};
+
+
+.. image:: img/cpp_static_array_set.png
+   :width: 400 px
+   :align: center
+   :target: http://www.cplusplus.com/doc/tutorial/arrays/
+
+
+
+Changing/Accessing Data
+-----------------------
+
+* Or we can assign/change and/or retrieve the values with indexing
+* These static arrays have known sizes at compile time, but we can still change the contents
+* In the below example
+    * The value at index ``3`` in ``foo`` will be set to ``99``
+    * The value from index ``3`` in ``foo`` (``99``) will be copied to ``x``
+
+.. code-block:: cpp
+    :linenos:
+
+    foo[3] = 99;
+    int x = foo[3];
+
+
+As Function Parameters
+----------------------
+
+* As you would expect, arrays can be passed to functions as parameters
+
+.. code-block:: cpp
+    :linenos:
+
+    void printArray(int toPrint[], int length){
+        for(int i = 0; i < length; i++){
+            std::cout << toPrint[i] << std::endl;
+        }
+    }
+
+
+.. warning::
+
+    These arrays are just chunks of RAM and do **not** know their own size; we must keep track of the array length on
+    our own. In the above example, notice how the ``length`` was also included as a parameter.
+
+    There is a way to get the length of an array, but it's a little cumbersome. The ``sizeof`` operator tells you how
+    much RAM a given value/type takes up. For example, if you ask ``sizeof(someInt)``, it will return the number of bytes
+    an integer needs (typically 4 bytes). Further, if you have an array ``int foo[5]``,  and ask ``sizeof(foo)``, it
+    will return the total amount of RAM that array uses (in this case, one would expect 20 bytes --- room for 5
+    integers, and an integer takes up 4 bytes). With this, I can calculate the size of the array with
+    ``sizeof(foo)/sizeof(foo[0])``.
+
+
+Arrays vs. Pointers
+-------------------
+
+* Arrays and pointers behave similarly
```

Comment:
This is potentially confusing. Arrays are references, not pointers. You can, however, dereference a pointer and offset it just like an array. The `[x]` operator actually means to dereference and offset by `x * sizeof(type)`, so it can apply to any referential type.

Maybe this topic just sucks.

Reply:
yuck

Reply:
I'll leave as is because I just don't care.